src/Analyzer.Core/TemplateAnalyzer.cs

```diff
@@ -99,9 +101,48 @@ public IEnumerable<IEvaluation> AnalyzeTemplate(string template, string template
                     throw new TemplateAnalyzerException(BicepCompileErrorMessage, e);
                 }
             }
+            var  evaluations = AnalyzeAllIncludedTemplates(template, template, parameters, templateFilePath, "", isBicep, sourceMap);
 
+            // For each rule we don't want to report the same line more than once
+            // This is a temporal fix
+            var evalsToValidate = new List<IEvaluation>();
+            var evalsToNotValidate = new List<IEvaluation>();
+            foreach (var eval in evaluations)
+            {
+                if (!eval.Passed && eval.Result != null)
+                {
+                    evalsToValidate.Add(eval);
+                }
+                else
+                {
+                    evalsToNotValidate.Add(eval);
+                }
+            }
+            var uniqueResults = new Dictionary<(string, int), IEvaluation>();
+            foreach (var eval in evalsToValidate)
+            {
+                uniqueResults.TryAdd((eval.RuleId, eval.Result.LineNumber), eval);
+            }
+            evaluations = uniqueResults.Values.Concat(evalsToNotValidate);
+
+            return evaluations;
+        }
+
+        /// <summary>
+        /// Analyzes ARM templates, recursively going through the nested templates
+        /// </summary>
+        /// <param name="rootTemplate">The original ARM Template JSON </param>
+        /// <param name="populatedTemplate">The ARM Template JSON with inherited parameters, variables, and functions, if applicable</param>
+        /// <param name="parameters">The parameters for the ARM Template JSON</param>
+        /// <param name="templateFilePath">The ARM Template file path</param>
+        /// <param name="pathPrefix">Prefix for resources' path used for line number matching</param>
+        /// <param name="isBicep">Whether this template was originally a Bicep file</param>
+        /// <param name="sourceMap">Source map that maps ARM JSON back to source Bicep</param>
+        /// <returns>An enumerable of TemplateAnalyzer evaluations.</returns>
+        private IEnumerable<IEvaluation> AnalyzeAllIncludedTemplates(string rootTemplate, string populatedTemplate, string parameters, string templateFilePath, string pathPrefix, bool isBicep, object sourceMap)
```

Comment:
I have no issues with isBicep and sourcemap but I think I might have to keep passing pathPrefix as a parameter since it is a prefix for the next template, not the current one. So I would need to construct the prefix, pass it as a parameter and then assign it to the context of the child. In that sense templateContext would, like you said, hold the context of the current template as opposed to holding that of the next one